pkg/proxy/service.go

```diff
@@ -17,17 +17,115 @@ limitations under the License.
 package proxy
 
 import (
+	"fmt"
+	"net"
 	"reflect"
+	"strings"
 	"sync"
 
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/record"
+	apiservice "k8s.io/kubernetes/pkg/api/service"
 	api "k8s.io/kubernetes/pkg/apis/core"
-	proxyutil "k8s.io/kubernetes/pkg/proxy/util"
+	"k8s.io/kubernetes/pkg/apis/core/helper"
+	utilproxy "k8s.io/kubernetes/pkg/proxy/util"
+	"k8s.io/kubernetes/pkg/util/ipv6"
 )
 
+// BaseServiceInfo contains base service information.
```

Comment:
I don't understand what it is is "base" for.  Can you beef up comments?  Same for Endpoint

Reply:
Sure, added more comments for clarification, PTAL, thanks!